src/web/app/pages-logs/logs-page.component.ts

```diff
@@ -88,45 +113,89 @@ export class LogsPageComponent implements OnInit {
       return;
     }
 
     this.hasResult = false;
     this.isSearching = true;
     this.searchResults = [];
     this.nextPageToken = '';
+    this.isFiltersExpanded = false;
     const localDateTime: Observable<number>[] = [
       this.resolveLocalDateTime(this.formModel.logsDateFrom, this.formModel.logsTimeFrom, 'Search period from'),
       this.resolveLocalDateTime(this.formModel.logsDateTo, this.formModel.logsTimeTo, 'Search period until'),
     ];
 
     forkJoin(localDateTime)
-        .pipe(
-            concatMap(([timestampFrom, timestampUntil]: number[]) => {
-              this.previousQueryParams = {
-                searchFrom: timestampFrom.toString(),
-                searchUntil: timestampUntil.toString(),
-                severities: Array.from(this.formModel.logsSeverity).join(','),
-              };
-              return this.logService.searchLogs(this.previousQueryParams);
-            }),
-            finalize(() => {
-              this.isSearching = false;
-              this.hasResult = true;
-            }))
-            .subscribe((generalLogs: GeneralLogs) => this.processLogs(generalLogs),
-              (e: ErrorMessageOutput) => this.statusMessageService.showErrorToast(e.error.message));
+      .pipe(
+        concatMap(([timestampFrom, timestampUntil]: number[]) => {
+          this.setQueryParams(timestampFrom, timestampUntil);
+          return this.logService.searchLogs(this.queryParams);
+        }),
+        finalize(() => {
+          this.isSearching = false;
+          this.hasResult = true;
+        }))
+      .subscribe((generalLogs: GeneralLogs) => this.processLogs(generalLogs),
+        (e: ErrorMessageOutput) => this.statusMessageService.showErrorToast(e.error.message));
+  }
+
+  private isFormValid(): boolean {
+    if (this.formModel.logsFilter === '') {
+      this.statusMessageService.showErrorToast('Please choose to filter by severity / minimum severity / event');
+      return false;
+    }
+    if (this.formModel.logsFilter === this.SEVERITY && this.formModel.logsSeverity === '') {
+      this.statusMessageService.showErrorToast('Please choose a severity level');
+      return false;
+    }
+    if (this.formModel.logsFilter === this.MIN_SEVERITY && this.formModel.logsMinSeverity === '') {
+      this.statusMessageService.showErrorToast('Please choose a minimum severity level');
+      return false;
+    }
+    if (this.formModel.logsFilter === this.EVENT && this.formModel.logsEvent === '') {
+      this.statusMessageService.showErrorToast('Please choose an event type');
+      return false;
+    }
+    if (!this.queryParams.sourceLocationFile && this.queryParams.sourceLocationFunction) {
+      this.isFiltersExpanded = true;
+      this.statusMessageService.showErrorToast('Please fill in Source location file or clear Source location function');
+      return false;
+    }
+
+    return true;
+  }
+
+  private setQueryParams(timestampFrom: number, timestampUntil: number): void {
```

Comment:
This method need to be renamed or better explained. Just looking at method name `setQueryParam` and two timestamp parameters tells nothing about some other fields being changed to `undefined` or other values.